packages/cli/src/lib/bundler/server.ts

```diff
@@ -123,7 +136,15 @@ export async function serveBundle(options: ServeOptions) {
     config: fullConfig,
     targetPath: paths.targetPath,
     watch() {
-      server?.invalidate();
+      if (server) {
+        if ('invalidate' in server) {
+          server.invalidate();
+        }
+
+        if ('restart' in server) {
+          server.restart();
+        }
+      }
```

Comment:
How about to create an extra function for this, as it is duplicated twice, same as on line `94-102`

Reply:
Could even separate the two:

```
webpackServer?.invalidate()
viteServer?.restart()
```